packages/sdk/src/clients/bridgePool/client.ts

```diff
@@ -0,0 +1,167 @@
+import { BridgePoolEthers, BridgePoolEthers__factory } from "@uma/contracts-node";
```

Comment:
It's not the exact same, but it seems that we're duplicating a lot of work by building bot clients in `financial-templates-lib` and then a frontend-focused client in the sdk that have a lot of repeated functionality (both primarily track state using events). See https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/clients/InsuredBridgeL1Client.ts.

@daywiss I know it's complicated to bridge the gap in these packages, but can you think of a _process_ s.t. we could unify these to take advantage of shared work? Maybe we could build all new clients in the sdk and import them in the bots?

Thoughts?

Reply:
yea this is a tough question. the financial templates clients are written with a specific purpose in mind and therefore have a lot of functionality built into them, much more than whats provided in the sdk.  One place though where there could be more alignment is around the shape of event data, ie having a common definition of the type of state that events should generate.  From there we could have a common event processor that could be shared between code bases, which is really the only value the sdk is providing at moment on contracts.

Anything more than that though, for instance if financial-templates relied on featureful clients from sdk, i would fear that it could create a bottleneck in the bot work.  If this is important enough though, i think we could put more effort into this.

Reply:
Even if one is a superset of the other or they're fairly disjoint, I think we could see real advantages from using the same underlying client and processing logic with the ability to turn on or off certain features or extend/compose the classes.

I think what we're doing now is "shipping the org chart", where the code reflects the organizational structure of those who wrote it. We should really avoid the tendency to write each application or feature completely fresh and re-use our existing work.

I think this is the same thing that has gotten us into trouble previously with price feeds where different applications support different types of price feeds because they didn't all use the same underlying implementation.

What's the urgency of this PR? Can we unify these now?